src/plugins/upf/upf_adf.c

```diff
@@ -522,10 +521,12 @@ vnet_upf_app_add_del (u8 * name, u32 flags, u8 add)
       }));
       /* *INDENT-ON* */
 
+      hash_unset_mem (sm->upf_app_by_name, app->name);
       upf_adf_remove (app->db_index);
       vec_free (app->name);
       hash_free (app->rules_by_id);
       pool_free (app->rules);
+      clib_memset (app, 0, sizeof (*app));
```

Comment:
is that memset truly needed?

Reply:
That's what other plugins are doing. I'd say it's not strictly needed but zeroing out objects before returning them to the pool may make attackers' life harder in some scenarios. This is probably a bit too far-fetched (if an attacker has access to this API he basically owns the VPP instance anyway), so if you think that's redundant, I can remove that.

Reply:
I've use something similar to catch bugs in the past. So its not totally redundant. I find it much more important to make sure we clear an entry when we get it from the pool.